tests/core/asyncio/test_connect.py

```diff
@@ -4,17 +4,16 @@
     generate_unique_name,
 )
 from lahja import (
+    AsyncioEndpoint,
     ConnectionAttemptRejected,
     ConnectionConfig,
-    Endpoint,
 )
 
 
 @pytest.mark.asyncio
 async def test_can_not_connect_conflicting_names():
-
     own = ConnectionConfig.from_name(generate_unique_name())
-    endpoint = Endpoint()
+    endpoint = AsyncioEndpoint()
```

Comment:
Not for this PR but in general: If the `TrioEndpoint` is API compatible to the `AsyncioEndpoint`. I guess instead of duplicating the tests we should be able to reuse much of them and just run them once with a `TrioEndpoint` and once with a `AsyncioEndpoint` right?

Reply:
@cburgdorf non-trivial to do with the way that the `pytest-asyncio` and `pytest-trio` mark tests functions, but in general, that would be ideal.  Assuming we plan to drop the `asyncio` implementations then I think it's fine to maintain them independent of each other.  I'm really not sure what the right thing to do is at this stage.  Thoughts?

Reply:
Ah, alright. If there's no simple solution, I'm ok with copying the tests.

>Assuming we plan to drop the asyncio implementations then I think it's fine to maintain them independent of each other. I'm really not sure what the right thing to do is at this stage. Thoughts?

I feel we should not drop the `AsyncioEndpoint` and here is why: I believe (and I may be biased here) that lahja provides one of the best solutions to do efficient and ergonomic multiprocess communication in the current Python landscape. Trinity aside, I believe it is a net win for the project to support both endpoints. Especially if we consider that `asyncio` is the *standard* solution for asynchronous IO in Python (even though `trio` seems to be the *better*  one).

Also, supporting both kind of endpoints is pretty nice for migrating a code base from `asyncio` to `trio`. Today, it's only us using this library but this may change in the future.

I think we can revisit this if maintaining the `AsyncioEndpoint` ever becomes too much of a burden but for now I'd prefer that we keep it.

Reply:
:+1: we can decide this later once we're done with it and see if maintaining it adds too much overhead.